main/src/ca/uwaterloo/flix/language/phase/Namer.scala

```diff
@@ -2065,4 +2065,21 @@ object Namer {
     def addTag(s: String, n: Name.QName, t: Name.Ident): UseEnv = copy(tags = tags + (s -> (n, t)))
   }
 
+  /**
+    * Merges the given `imports` into the given import environment `ienv0`.
+    */
+  private def mergeImportEnvs(imports: List[WeededAst.Import], ienv0: ImportEnv): Validation[ImportEnv, NameError] = {
+    Validation.fold(imports, ienv0) {
+      case (ienv1, WeededAst.Import.Import(name, alias, loc)) =>
+        ienv1.get(alias.name) match {
+          case None => (ienv1 + (alias.name -> name)).toSuccess
+          case Some(otherName) => mkDuplicateNamePair(alias.name, loc, SourceLocation.mk(otherName.sp1, otherName.sp2))
+        }
+    }
+  }
+
+  /**
+    * Represents an environment of "imported" Java classes or interfaces
+    */
+  private type ImportEnv = Map[String, Name.JavaName]
```

Comment:
@magnus-madsen doesn't usually like type aliases :)

Reply:
Indeed. We try to avoid type aliases.